imagebuildah/build.go

```diff
@@ -280,7 +280,7 @@ func preprocessDockerfileContents(r io.Reader, ctxDir string) (rdrCloser *io.Rea
 	stdout := bytes.Buffer{}
 	stderr := bytes.Buffer{}
 
-	cmd := exec.Command(cppPath, "-E", "-iquote", ctxDir, "-")
+	cmd := exec.Command(cppPath, "-E", "-iquote", "-traditional", "-undef", ctxDir, "-")
```

Comment:
The new options can't come between `-iquote` and the directory name. https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html#Directory-Options